packages/models/src/utils/decorator/provider.ts

```diff
@@ -37,15 +63,84 @@ export function Pattern({
   cellPattern,
   schemaPattern,
 }: {
-  cellPattern?: CellPattern
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  cellPattern?: (...args: Array<any>) => CellPattern
   schemaPattern?: SchemaPattern
 }): ClassDecorator {
   return function (target) {
     if (cellPattern) {
-      Reflect.defineMetadata(ProviderKey.CellPattern, cellPattern, target)
+      const patterns = Reflect.getMetadata(ProviderKey.CellPattern, target) ?? []
+      patterns.push(cellPattern)
+      Reflect.defineMetadata(ProviderKey.CellPattern, patterns, target)
     }
     if (schemaPattern) {
       Reflect.defineMetadata(ProviderKey.SchemaPattern, schemaPattern, target)
     }
   }
 }
+
+export function DataCellPattern(data: string): ClassDecorator {
+  const cellPattern = (obj: { lockScript: Script }) => {
+    return (value: UpdateStorageValue) => {
+      const cellLock = value.cell.cellOutput.lock
+      return (
+        cellLock.args === obj.lockScript?.args &&
+        cellLock.codeHash === obj.lockScript?.codeHash &&
+        cellLock.hashType === obj.lockScript?.hashType &&
+        value.cell.data === data
+      )
+    }
+  }
+
+  return Pattern({ cellPattern })
+}
+
+export function DataPrefixCellPattern(prefix: string): ClassDecorator {
+  const cellPattern = (obj: { lockScript: Script }) => {
+    return (value: UpdateStorageValue) => {
+      const cellLock = value.cell.cellOutput.lock
+      return (
+        cellLock.args === obj.lockScript?.args &&
+        cellLock.codeHash === obj.lockScript?.codeHash &&
+        cellLock.hashType === obj.lockScript?.hashType &&
```

Comment:
Why does `DataPrefixCellPattern` has checking of `lockScript`

Reply:
> Why does `DataPrefixCellPattern` has checking of `lockScript`

This has not been finished yet, I'll push the changes after test.